src/Resolvers/Microsoft.NET.Sdk.WorkloadManifestReader/SdkDirectoryWorkloadManifestProvider.GlobalJsonReader.cs

```diff
@@ -0,0 +1,180 @@
+using static Microsoft.NET.Sdk.WorkloadManifestReader.WorkloadManifestReader;
+using System.Runtime.Serialization;
+
+#if USE_SYSTEM_TEXT_JSON
+using System.Text.Json;
+#else
+using Newtonsoft.Json;
+using JsonTokenType = Newtonsoft.Json.JsonToken;
+#endif
+
+namespace Microsoft.NET.Sdk.WorkloadManifestReader
+{
+    public partial class SdkDirectoryWorkloadManifestProvider
+    {
+        static class GlobalJsonReader
+        {
+            public static string? GetWorkloadVersionFromGlobalJson(string globalJsonPath)
+            {
+                if (string.IsNullOrEmpty(globalJsonPath))
+                {
+                    return null;
+                }
+
+                using var fileStream = File.OpenRead(globalJsonPath);
+
+#if USE_SYSTEM_TEXT_JSON
+                var readerOptions = new JsonReaderOptions
+                {
+                    AllowTrailingCommas = true,
+                    CommentHandling = JsonCommentHandling.Skip
+                };
+                var reader = new Utf8JsonStreamReader(fileStream, readerOptions);
+#else
+                using var textReader = new StreamReader(fileStream, System.Text.Encoding.UTF8, true);
+                using var jsonReader = new JsonTextReader(textReader);
+
+                var reader = new Utf8JsonStreamReader(jsonReader);
+#endif
+
+                string? workloadVersion = null;
+
+                ConsumeToken(ref reader, JsonTokenType.StartObject);
+                while (reader.Read())
+                {
+                    switch (reader.TokenType)
+                    {
+                        case JsonTokenType.PropertyName:
+                            var propName = reader.GetString();
+                            if (string.Equals("sdk", propName, StringComparison.OrdinalIgnoreCase))
+                            {
+                                ConsumeToken(ref reader, JsonTokenType.StartObject);
+
+                                bool readingSdk = true;
+                                while (readingSdk && reader.Read())
+                                {
+                                    switch (reader.TokenType)
+                                    {
+                                        case JsonTokenType.PropertyName:
+                                            var sdkPropName = reader.GetString();
+                                            if (string.Equals("workloadVersion", sdkPropName, StringComparison.OrdinalIgnoreCase))
+                                            {
+                                                workloadVersion = ReadString(ref reader);
+                                            }
+                                            else
+                                            {
+                                                ConsumeValue(ref reader);
+                                            }
+                                            break;
+                                        case JsonTokenType.EndObject:
```

Comment:
Would this cause a failure if someone was using the `msbuild-sdks` global.json option? It seems that would be an object, so we may need to handle the `JsonToken.TypeStartObject`

Reply:
The StartObject token would need to come after a PropertyName token, and should be handled by `ConsumeValue`.  The tests include global.json files with `msbuild-sdks` in them.  I'm not an expert at json parsing but I copied the patterns from the existing workload manifest parser, so I think the global.json parser is probably correct.